peas/weather_abstract.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+
+import logging
+import numpy as np
+import re
+import serial
+import sys
+import time
+import requests
+
+from datetime import datetime as dt
+from dateutil.parser import parse as date_parser
+
+import astropy.units as u
+from astropy.units import cds
+from astropy.table import Table
+from astropy.time import Time, TimeISO, TimeDelta
+
+from pocs.utils.messaging import PanMessaging
+
+from . import load_config
+from .PID import PID
+
+def get_mongodb():
+    from pocs.utils.database import PanMongo
+    return PanMongo()
+
+# -----------------------------------------------------------------------------
+#   Base class to read and check weather data
+# -----------------------------------------------------------------------------
+class WeatherAbstract(object):
+
+    """ Base class for checking generic weather data and sending it to the
+        required location.
+    """
+
+    def __init__(self, use_mongo=True):
+        self.config = load_config()
+
+        self.thresholds = self.config.get('thresholds')
+        self.safety_delay = self.config.get('safety_delay', 15.)
+
+        self.db = None
+        if use_mongo:
+            self.db = get_mongodb()
+
+        self.messaging = None
+        self.safe_dict = None
+        self.weather_entries = list()
+
+    def send_message(self, msg, channel='weather'):
+        if self.messaging is None:
+            self.messaging = PanMessaging.create_publisher(6510)
+
+        self.messaging.send_message(channel, msg)
+
+    def capture(self, data):
+        # Make Safety Decision
+        self.safe_dict = self.make_safety_decision(data)
+
+        data['Safe'] = self.safe_dict['Safe']
+        data['Sky condition'] = self.safe_dict['Sky']
+        data['Wind condition'] = self.safe_dict['Wind']
+        data['Gust condition'] = self.safe_dict['Gust']
+        data['Rain condition'] = self.safe_dict['Rain']

```

Comment:
These 5 lines are just duplicating the contents of the dictionary returned by `make_safety_decision()` with the addition of the word 'condition'.  Just put 'condition' into the names in `self._safety_methods`  in `__init__()` and remove all this.

Reply:
Also, reuse of the variable `data`, 1st as the current weather data values passes as an argument to this method, then as dictionary of safety status info returned from `make_safety_decision()`.  That's just confusing, use a different variable name for the second case, e.g. `safety_status`.

Reply:
And to continue my nitpicking, `data` is a terrible variable name. Far too vague. Variable (& function, & class) names should describe the meaning (not implementation) of the thing in question.

Reply:
Yep, I added condition into the names.
Mmm, I didn't catch on to that before... Makes it much more confusing! I have fixed it up, looks like this:
`    def capture(self, current_values):
        # Make Safety Decision and store current weather
        current_weather = self.make_safety_decision(current_values)
        self.weather_entries.append(current_weather)

        if send_message:
            self.send_message({'data': current_weather}, channel='weather')

        if use_mongo:
            self.db.insert_current('weather', current_weather)

        return current_weather`